### DISCUSSION TITLE
Clear Button of elements in multi mode of Typeahead is rendered underneath the label

### DISCUSSION THREAD

**Discussion by MichalskiMaximilian** [ericgio/react-bootstrap-typeahead#856] [Q&A] [answered]
While im implementing a Typeahead with no special settings and with introduction of the examples page https://ericgio.github.io/react-bootstrap-typeahead/, the clear button is rendered underneath the element. Taking a look into the devtools of my browser i saw the there is a div with the selected element and then comes the button. 

Is there a possibility to avoid this? 

![image](https://github.com/ericgio/react-bootstrap-typeahead/assets/144446877/25075f4d-8eb9-43a7-afce-b835eb166ec7)

![image](https://github.com/ericgio/react-bootstrap-typeahead/assets/144446877/cbcce203-07f5-4445-ac38-c8a1bcffd46f)

Kind regards Max

**Comment by ericgio** [comment]
~~The simplest approach would be to use the [`renderToken`](https://github.com/ericgio/react-bootstrap-typeahead/blob/main/docs/Rendering.md#rendertokenoption-objectstring-props-object-index-number) prop to customize rendering of the token. A more complex approach would be to use `renderInput` to [customize the input rendering](https://codesandbox.io/p/sandbox/misty-water-u43sf) as well as the token.~~

Edit: I misunderstood the problem. This is a CSS issue, not a DOM customization issue.

**Comment by MichalskiMaximilian** [comment]
Unfortunatly is the result after the implementation of the renderToken or the renderInput the same. Is there some i made wrong i forgot? 

![image](https://github.com/ericgio/react-bootstrap-typeahead/assets/144446877/7a96d304-eaf9-4350-bb09-d674684fc744)


```ts
<Typeahead 
    id="buildings" 
    multiple={true} 
    size={'sm'} 
    options={buildings.sort((a, b) => a.idNum.localeCompare(b.idNum))} 
    placeholder={CodeStrings.InvestigationOrder.ChooseBuilding} 
    selected={selectedBuildings}
    onChange={(option: any) => setSelectedBuildings(option)}
    labelKey={(option: any) => `${option.idNum} ( ${option.economicUnit} )`}
    emptyLabel={CodeStrings.NoResults}
    renderToken={(option: any, props, idx) => (
        <Token
            tabIndex={idx}
            option={option}
            multiple={true}
            disabled={false}
            onRemove={props.onRemove}>
            {`${option.idNum} ( ${option.economicUnit} )`}
        </Token>
    )}
/>
```

```ts

<Typeahead
    id="buildings"
    multiple
    onChange={(option: any) => setSelectedBuildings(option)}
    options={buildings}
    placeholder="Choose a state..."
    labelKey={(option: any) => `${option.idNum}`}
    renderInput={(inputProps, props) => (
        <TypeaheadInputMulti {...inputProps} selected={selectedBuildings}>
            {selectedBuildings.map((option: any, idx) => (
                <Token
                    index={idx}
                    key={option.idNum}
                    onRemove={props.onRemove}
                    option={option}>
                    {option.idNum}
                </Token>
            ))}
        </TypeaheadInputMulti>
    )}
    selected={selectedBuildings}
/>
```

**Comment by ericgio** [comment, marked as answer]
@MichalskiMaximilian sorry, I totally misunderstood your original issue. It looks like you're not including the CSS in your project, so the token is rendering incorrectly. Please see the section on [including CSS](https://github.com/ericgio/react-bootstrap-typeahead/tree/main?tab=readme-ov-file#css) in the docs.

**Reply by MichalskiMaximilian** [reply]
This looks pretty good and solves my problem. Thank you very much for your help and the quick responses.
